src/PhpDocument.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+namespace LanguageServer;
+
+use \LanguageServer\Protocol\{Diagnostic, DiagnosticSeverity, Range, Position, SymbolKind, TextEdit};
+
+use PhpParser\{Error, Comment, Node, ParserFactory, NodeTraverser, Lexer, Parser};
+use PhpParser\PrettyPrinter\Standard as PrettyPrinter;
+use PhpParser\NodeVisitor\NameResolver;
+
+class PhpDocument
+{
+    private $stmts;
+    private $client;
+    private $project;
+    private $symbols = [];
+    private $parser;
+    private $uri;
+
+    public function __construct(string $uri, Project $project, LanguageClient $client, Parser $parser)
+    {
+        $this->uri = $uri;
+        $this->project = $project;
+        $this->client = $client;        
+        $this->parser = $parser;
+    }
+
+    /**
+     * Returns all symbols in this document.
+     *
+     * @return SymbolInformation[]
+     */
+    public function getSymbols()
+    {
+        return $this->symbols;
+    }
+
+    /**
+     * Returns symbols in this document filtered by query string.
+     *
+     * @param string $query The search query
+     * @return SymbolInformation[]
+     */
+    public function findSymbols(string $query)
+    {
+        return array_filter($this->symbols, function($symbol) use(&$query) {
+            return stripos($symbol->name, $query) !== false;
+        });
+    }
+
+    /**
+     * Re-parses a source file, updates the AST and reports parsing errors that may occured as diagnostics
+     *
+     * @param string $content The new content of the source file
+     * @return void
+     */
+    public function updateAst(string $content)
+    {
+        $stmts = null;
+        try {
+            $stmts = $this->parser->parse($content);
+        }
+        catch(Error $e) {
+            // Parser still throws errors. e.g for unterminated comments

```

Comment:
I found this weird as well. I had the extension crash multiple times while writing a block comment.
But you are right, we should probably catch this specific extension and add it as a parse error.